### ISSUE TITLE
Some linux systems may need CXXFLAGS+=-std=c++0x

### ISSUE THREAD

**Issue by justintime4tea** [nagisa/rust_tracy_client#18]
Other Linux users may run into this; I am uncertain. I was having trouble using any of the tracy rust crates because it was failing to compile the underlying TracyClient.cpp and friends.  Perhaps we can add something to the documentation regarding maybe needing special CXXFLAGS? or perhaps CXXFLAG can conditionally be added when seemingly necessary and running cargo build?

```
cargo build
// ... lots of output
  error occurred: Command "c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-DTRACY_ENABLE" "-o" "target/debug/build/tracy-client-sys-6734926446c3849b/out/tracy/TracyClient.o" "-c" "tracy/TracyClient.cpp" with args "c++" did not execute successfully (status code exit status: 1).
```

When running c++ manually the output is many lines like this:

```
c++ -O0 -ffunction-sections -fdata-sections -fPIC -g -fno-omit-frame-pointer -m64 -DTRACY_ENABLE -o target/debug/build/tracy-client-sys-67123123b/out/tracy/TracyClient.o -c tracy/TracyClient.cpp

.cargo/registry/src/github.com-1ecc6299db9ec823/tracy-client-sys-0.16.0/tracy/client/TracyDxt1.cpp:633:24: error: ‘c’ does not name a type
             const auto c = ProcessRGB( (uint8_t*)buf );
                        ^
```

It appears as though the "auto" keyword (from C++11?) is not available by default when using g++? (I'm not a C person).

Apparently the issue was I was compiling with g++ (may be diff for others?) and to enable C++11 features it must be passed as a flag. The "standard" value is "c++0x" for c++11. When you add that to CXXFLAGS cargo build compilation can continue. 

The complete cargo build command:
`CXXFLAGS=-std=c++0x cargo build`

**Comment by nagisa** [comment]
Tracy requires a compiler that supports gnu++17 (C++ 2017 with GNU extensions) language standard as per [this](https://github.com/wolfpld/tracy/blob/d71d76414b8cc6a3d5658ceed57e540687d1c23e/library/unix/build.mk#L2).

[We set this flag](https://github.com/nagisa/rust_tracy_client/blob/main/tracy-client-sys/build.rs#L38) if it is supported. I suspect that your toolchain is too old to support this, which is why it ended up not being set.

**Comment by justintime4tea** [comment]
Interestingly it compiles fine with C++11, are there specific features of > 11 in use? I haven't seem to run into any errors, segfaults or bugs. 

Would be nice to lower the requirements if they aren't strict or are arbitrary requirements. A middle ground might be automatic detection of latest supported c++ on the system and at the very least something in the Docs that mentions of what C++ >=# is strictly/actually required.

**Comment by nagisa** [comment]
I don't think you will run into issues, but given how the upstream build script is written, features from the more recent standards can crop in at any release. I think it would make sense to ask upstream how attached are they to gnu++17 in their client libraries and if not at all, perhaps lowering it to `c++11` would be possible. Want to do it or should I?

**Comment by justintime4tea** [comment]
That makes a lot of sense! I wouldn't mind reaching out to upstream. Is there a contact you know or mailing list or do you think opening a "question issue" would be appropriate?

**Comment by nagisa** [comment]
An issue on https://github.com/wolfpld/tracy is fine.

**Comment by wolfpld** [comment]
https://github.com/wolfpld/tracy/commit/d6835f3a7ae1bdc60fef8a2d039931e301181c87
